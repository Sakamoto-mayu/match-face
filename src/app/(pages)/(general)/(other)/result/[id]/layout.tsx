export const metadata = {
  title: "回答確認",
  description: "一般ユーザー回答確認ページ",
};

export default function ResultLayout({
  children,
}: {
  children: React.ReactNode;
}) {
  return <>{children}</>;
}
