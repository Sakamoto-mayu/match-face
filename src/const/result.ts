export const Answer = [
  {
    user_id: "user1",
    project_id: "1",
    question_id: "1",
    answer_date: new Date("2023-07-06"),
    content:
      "回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。",
  },
  {
    user_id: "user1",
    project_id: "1",
    question_id: "2",
    answer_date: new Date("2023-07-06"),
    content: "選択肢A",
  },
  {
    user_id: "user1",
    project_id: "1",
    question_id: "3",
    answer_date: new Date("2023-07-06"),
    content:
      "回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。",
  },
  {
    user_id: "user1",
    project_id: "2",
    question_id: "1",
    answer_date: new Date("2023-07-06"),
    content:
      "回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。",
  },
  {
    user_id: "user1",
    project_id: "2",
    question_id: "2",
    answer_date: new Date("2023-07-06"),
    content:
      "回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。",
  },
  {
    user_id: "user1",
    project_id: "2",
    question_id: "3",
    answer_date: new Date("2023-07-06"),
    content:
      "回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。",
  },
  {
    user_id: "user1",
    project_id: "3",
    question_id: "1",
    answer_date: new Date("2023-07-06"),
    content:
      "回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。",
  },
  {
    user_id: "user1",
    project_id: "3",
    question_id: "2",
    answer_date: new Date("2023-07-06"),
    content:
      "回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。",
  },
  {
    user_id: "user1",
    project_id: "3",
    question_id: "3",
    answer_date: new Date("2023-07-06"),
    content:
      "回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。回答内容が入ります。",
  },
];

export const ModelAnswer = [
  {
    editor_id: "admin1",
    project_id: "1",
    question_id: "1",
    edit_date: new Date("2023-07-07"),
    content:
      "模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。",
  },
  {
    editor_id: "admin1",
    project_id: "1",
    question_id: "2",
    edit_date: new Date("2023-07-07"),
    content:
      "模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。",
  },
  {
    editor_id: "admin1",
    project_id: "1",
    question_id: "3",
    edit_date: new Date("2023-07-07"),
    content:
      "模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。",
  },
  {
    editor_id: "admin1",
    project_id: "2",
    question_id: "1",
    edit_date: new Date("2023-07-07"),
    content:
      "模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。",
  },
  {
    editor_id: "admin1",
    project_id: "2",
    question_id: "2",
    edit_date: new Date("2023-07-07"),
    content:
      "模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。",
  },
  {
    editor_id: "admin1",
    project_id: "2",
    question_id: "3",
    edit_date: new Date("2023-07-07"),
    content:
      "模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。",
  },
  {
    editor_id: "admin1",
    project_id: "3",
    question_id: "1",
    edit_date: new Date("2023-07-07"),
    content:
      "模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。",
  },
  {
    editor_id: "admin1",
    project_id: "3",
    question_id: "2",
    edit_date: new Date("2023-07-07"),
    content:
      "模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。",
  },
  {
    editor_id: "admin1",
    project_id: "3",
    question_id: "3",
    edit_date: new Date("2023-07-07"),
    content:
      "模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。模範解答が入ります。",
  },
];

export const Comment = [
  {
    user_id: "user1",
    admin_id: "adminuser",
    project_id: "1",
    comment_date: new Date("2023-07-07"),
    content:
      "コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。",
  },
  {
    user_id: "user1",
    admin_id: "adminuser",
    project_id: "2",
    comment_date: new Date("2023-07-07"),
    content:
      "コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入りますコメント内容が入ります。",
  },
  {
    user_id: "user1",
    admin_id: "adminuser",
    project_id: "3",
    comment_date: new Date("2023-07-07"),
    content:
      "コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入りますコメント内容が入ります。",
  },
];

export const User = [
  {
    id: "adminuser",
    name: "管理者太郎",
    image_url: "/icon/human_icon.png",
  },
];
