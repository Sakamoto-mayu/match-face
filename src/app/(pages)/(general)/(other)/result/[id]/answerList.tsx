import { question } from "@/const/testing";
import { Answer } from "@/const/result";
import ModelAnswerContent from "@/components/pages/general/result/ModelAnswer";

type Props = {
  project_id: number;
};

export const AnswerList = ({ project_id }: Props) => {
  const currentQuestion = question.filter(
    (question) => question.project_id === project_id
  );

  const currentAnswer = Answer.filter(
    (answer) => answer.project_id === project_id
  );

  return (
    <>
      {currentQuestion.map((question, index) => (
        <div key={index} className="py-5">
          <h3 className="text-xl">
            Q{index + 1}:{question.name}
          </h3>
          <p id={`content-${index + 1}`} className="mt-2 mb-4">
            {question.content}
          </p>
          {question.type === "writing" ? (
            <div
              id={`answer-${index + 1}`}
              className="border border-gray p-3 mt-2 rounded-md shadow-md"
            >
              <h3 className="mb-2 text-xl">あなたの回答</h3>
              {currentAnswer[index].content}
            </div>
          ) : (
            <>
              <ul>
                {question.choices?.map((choice, index) => (
                  <li key={index}>・{choice}</li>
                ))}
              </ul>
              <div
                id={`answer-${index + 1}`}
                className="border border-gray p-3 mt-5 rounded-md shadow-md"
              >
                <h3 className="mb-2 text-xl">あなたの回答</h3>
                {currentAnswer[index].content}
              </div>
            </>
          )}
          <ModelAnswerContent
            project_id={project_id}
            question_id={question.id}
          />
        </div>
      ))}
    </>
  );
};
