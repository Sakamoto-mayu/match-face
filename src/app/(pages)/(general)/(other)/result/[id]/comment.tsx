import TextArea from "@/components/ui/TextArea";
import { Comment } from "@/const/result";
import { User } from "@/const/result";
import Image from "next/image";

type Props = {
  user_id: number;
  project_id: number;
};

export const CommentContent = ({ user_id, project_id }: Props) => {
  const currentComment = Comment.filter(
    (comment) => comment.user_id === user_id && project_id === project_id
  );
  const commentUser = User.filter(
    (user) => user.id === currentComment[0].admin_id
  );

  return (
    <>
      <h3 className="text-xl mb-3">{commentUser[0].name}からのコメント</h3>
      <div className="border border-black rounded-md p-3 flex justify-between ">
        <div className="w-1/6 flex justify-center items-center">
          <Image
            src={commentUser[0].image_url}
            alt="コメントしたユーザーのアイコン"
            width={70}
            height={70}
            priority
          />
        </div>
        <div id="comment" className="w-5/6">
          {currentComment[0].content}
        </div>
      </div>
    </>
  );
};
