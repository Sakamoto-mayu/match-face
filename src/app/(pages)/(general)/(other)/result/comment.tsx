import TextArea from "@/components/ui/TextArea";
import { Comment } from "@/const/result";
import Image from "next/image";

type Props = {
  user_id: string;
  project_id: number;
};

export const CommentContent = ({ user_id, project_id }: Props) => {
  const currentComment = Comment.filter(
    (comment) => comment.user_id === user_id && project_id === project_id
  );

  return (
    <>
      <h3 className="text-xl mb-3">〇〇からのコメント</h3>
      <div className="border border-black rounded-md p-3 flex justify-between ">
        <div className="w-1/6 flex justify-center items-center">
          <Image
            src={"/icon/human_icon.png"}
            alt="コメントしたユーザーのアイコン"
            width={70}
            height={70}
          />
        </div>
        <div className="w-5/6">{currentComment[0].content}</div>
      </div>
    </>
  );
};
