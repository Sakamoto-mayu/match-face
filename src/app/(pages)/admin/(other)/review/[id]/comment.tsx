import TextArea from "@/components/ui/TextArea";
import { User } from "@/const/review";
import Image from "next/image";
import { useState } from "react";

type Props = {
  admin_id: number;
};

export const CommentContent = ({ admin_id }: Props) => {
  const commentUser = User.filter((user) => user.id === admin_id);
  const [comment, setComment] = useState("");

  return (
    <>
      <label htmlFor="comment">
        <h3 className="text-xl mb-3">▶︎コメントを入力する</h3>
      </label>
      <div className=" flex justify-between ">
        <div className="w-1/6 flex justify-center items-center">
          <Image
            src={commentUser[0].image_url}
            alt="コメントしたユーザーのアイコン"
            width={70}
            height={70}
          />
        </div>
        <TextArea
          id="comment"
          cols={50}
          rows={6}
          value={comment}
          onChange={(e) => {
            setComment(e.target.value);
          }}
        />
      </div>
    </>
  );
};
