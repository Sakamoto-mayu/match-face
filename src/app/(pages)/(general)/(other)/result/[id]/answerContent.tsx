"use client";

import { project } from "@/const/testing";
import { AnswerList } from "./answerList";
import { CommentContent } from "./comment";
import OrangeButton from "@/components/ui/button/OrangeButton";
import { Answer } from "@/const/result";
import { useRouter } from "next/navigation";

type Props = {
  user_id: number;
  project_id: number;
};

export const AnswerContent = ({ user_id, project_id }: Props) => {
  const router = useRouter();
  const currentProject = project.filter(
    (project) => project.id === Number(project_id)
  );
  const currentAnswer = Answer.filter(
    (answer) => answer.project_id === Number(project_id)
  );

  return (
    <div className="w-1/2 py-5">
      <div className="flex justify-between items-center my-3">
        <h1 className="text-2xl">回答の確認</h1>
        {/*現状のデータでは4以降の解答データを呼び出そうとするとエラーになります */}
        <p>回答日時:{currentAnswer[0].answer_date.toLocaleDateString()}</p>
      </div>
      <hr className="border border-black" />
      <h2 className="text-xl my-2">案件名：{currentProject[0].name}</h2>
      <p className="text-lg">{currentProject[0].project_detail}</p>
      <AnswerList project_id={project_id} />
      <CommentContent user_id={user_id} project_id={project_id} />
      <div className="flex justify-around my-20">
        <OrangeButton
          className="border rounded-none"
          label="質問一覧へ"
          onClick={() => router.push("/questions")}
        />
        <OrangeButton
          className="border rounded-none"
          label="回答履歴一覧へ"
          onClick={() => router.push("/histories")}
        />
      </div>
    </div>
  );
};
