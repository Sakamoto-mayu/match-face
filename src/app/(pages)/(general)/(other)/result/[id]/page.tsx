import { AnswerContent } from "./answerContent";

const ResultPage = ({ params }: { params: { id: number } }) => {
  const user_id = 1;

  return (
    <div className="flex flex-col items-center">
      <AnswerContent user_id={user_id} project_id={params.id} />
    </div>
  );
};

export default ResultPage;
