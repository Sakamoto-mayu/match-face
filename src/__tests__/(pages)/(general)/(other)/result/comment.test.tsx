import { CommentContent } from "@/app/(pages)/(general)/(other)/result/[id]/comment";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { Comment } from "@/const/result";

const testData = {
  user_id: 1,
  project_id: 1,
  admin_user: "管理者太郎",
  img_url: "%2Ficon%2Fhuman_icon.png",
  comment: Comment[0],
};

describe("Commentコンポーネントテスト", () => {
  test("管理者の名前が表示される", () => {
    render(
      <CommentContent
        user_id={testData.user_id}
        project_id={testData.project_id}
      />
    );

    const heading = screen.getByRole("heading", {
      name: "管理者太郎からのコメント",
    });
    expect(heading).toBeInTheDocument();
  });

  test("管理者のアイコンが表示される", () => {
    render(
      <CommentContent
        user_id={testData.user_id}
        project_id={testData.project_id}
      />
    );

    //srcのURL指定方法については要検討
    const icon = screen.getByRole("img");
    expect(icon.getAttribute("src")).toContain(testData.img_url);
  });

  test("コメント内容が表示される", () => {
    render(
      <CommentContent
        user_id={testData.user_id}
        project_id={testData.project_id}
      />
    );

    const commentContents = document.querySelector("#comment");
    expect(commentContents).toHaveTextContent(testData.comment.content);
  });

  test("スナップショットテスト", () => {
    const contents = render(
      <CommentContent
        user_id={testData.user_id}
        project_id={testData.project_id}
      />
    );

    expect(contents).toMatchSnapshot();
  });
});
