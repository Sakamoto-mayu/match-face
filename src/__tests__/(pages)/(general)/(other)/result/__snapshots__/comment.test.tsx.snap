// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Commentコンポーネントテスト スナップショットテスト 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h3
        class="text-xl mb-3"
      >
        管理者太郎
        からのコメント
      </h3>
      <div
        class="border border-black rounded-md p-3 flex justify-between "
      >
        <div
          class="w-1/6 flex justify-center items-center"
        >
          <img
            alt="コメントしたユーザーのアイコン"
            data-nimg="1"
            decoding="async"
            fetchpriority="high"
            height="70"
            src="/_next/image?url=%2Ficon%2Fhuman_icon.png&w=256&q=75"
            srcset="/_next/image?url=%2Ficon%2Fhuman_icon.png&w=96&q=75 1x, /_next/image?url=%2Ficon%2Fhuman_icon.png&w=256&q=75 2x"
            style="color: transparent;"
            width="70"
          />
        </div>
        <div
          class="w-5/6"
          id="comment"
        >
          コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <h3
      class="text-xl mb-3"
    >
      管理者太郎
      からのコメント
    </h3>
    <div
      class="border border-black rounded-md p-3 flex justify-between "
    >
      <div
        class="w-1/6 flex justify-center items-center"
      >
        <img
          alt="コメントしたユーザーのアイコン"
          data-nimg="1"
          decoding="async"
          fetchpriority="high"
          height="70"
          src="/_next/image?url=%2Ficon%2Fhuman_icon.png&w=256&q=75"
          srcset="/_next/image?url=%2Ficon%2Fhuman_icon.png&w=96&q=75 1x, /_next/image?url=%2Ficon%2Fhuman_icon.png&w=256&q=75 2x"
          style="color: transparent;"
          width="70"
        />
      </div>
      <div
        class="w-5/6"
        id="comment"
      >
        コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。コメント内容が入ります。
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
